packages/master-detail-layout/src/vaadin-master-detail-layout.js

```diff
@@ -0,0 +1,166 @@
+/**
+ * @license
+ * Copyright (c) 2025 - 2025 Vaadin Ltd.
+ * This program is available under Apache License Version 2.0, available at https://vaadin.com/license/
+ */
+import { css, html, LitElement } from 'lit';
+import { defineCustomElement } from '@vaadin/component-base/src/define.js';
+import { ElementMixin } from '@vaadin/component-base/src/element-mixin.js';
+import { PolylitMixin } from '@vaadin/component-base/src/polylit-mixin.js';
+import { ThemableMixin } from '@vaadin/vaadin-themable-mixin/vaadin-themable-mixin.js';
+
+/**
+ * `<vaadin-master-detail-layout>` is a web component for building UIs with a master
+ * (or primary) area and a detail (or secondary) area that is displayed next to, or
+ * overlaid on top of, the master area, depending on configuration and viewport size.
+ *
+ * @customElement
+ * @extends HTMLElement
+ * @mixes ThemableMixin
+ * @mixes ElementMixin
+ */
+class MasterDetailLayout extends ElementMixin(ThemableMixin(PolylitMixin(LitElement))) {
+  static get is() {
+    return 'vaadin-master-detail-layout';
+  }
+
+  static get styles() {
+    return css`
+      :host {
+        display: flex;
+        box-sizing: border-box;
+        height: 100%;
+      }
+
+      :host([hidden]) {
+        display: none !important;
+      }
+
+      :host(:not([has-detail])) [part='detail'] {
+        display: none;
+      }
+
+      /* No fixed size */
+      :host(:not([has-master-size])) [part='master'] {
+        flex-grow: 1;
+        flex-basis: var(--_master-min-size, 50%);
+      }
+
+      :host(:not([has-detail-size])) [part='detail'] {
+        flex-grow: 1;
+        flex-basis: var(--_detail-min-size, 50%);
+      }
+
+      /* Fixed size */
+      :host([has-master-size]) [part='master'] {
+        width: var(--_master-size);
+      }
+
+      :host([has-detail-size]) [part='detail'] {
+        width: var(--_detail-size);
+      }
+    `;
+  }
+
+  static get properties() {
+    return {
+      /**
+       * Fixed size (in CSS length units) to be set on the detail pane.
+       * When specified, it prevents the detail pane from growing.
+       *
+       * @attr {string} detail-size
+       */
+      detailSize: {
```

Comment:
Is there a reason to prefer `size` to `width` in the name?

Reply:
This is in order to support vertical orientation (to be added later).

Reply:
Thanks, got it.